### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by typaclever**

### CODE REVIEW

File: src/main/java/model/user/User.java [hunk 1]
```diff
@@ -3,17 +3,24 @@
...
 public class User {
     @Id
-    @GeneratedValue(strategy = GenerationType.IDENTITY)
-    private int id;
-    private String logIn;
-    private String password;
+    String logIn;
+    private byte[] salt;
+    private byte[] hashPassword;
     @OneToMany(cascade = CascadeType.PERSIST, mappedBy = "user", fetch = FetchType.LAZY)
```
Comment (AndrewPolishchuc) [inline src/main/java/model/user/User.java]: Why OneToMany? What about many to many?

File: src/main/java/service/connection/impl/ConnectionServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import model.connection.Connection;
+import repository.connection.ConnectionRepository;
+import repository.connection.impl.ConnectionRepositoryImpl;
+import service.connection.ConnectionService;
+
+import java.util.List;
+
+public class ConnectionServiceImpl implements ConnectionService {
+    private ConnectionRepository connectionRepository = new ConnectionRepositoryImpl();
```
Comment (AndrewPolishchuc) [inline src/main/java/service/connection/impl/ConnectionServiceImpl.java]: Bad practice! Read about java autowired!

File: src/main/java/service/display/impl/DisplayServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import model.display.Display;
+import repository.display.DisplayRepository;
+import repository.display.impl.DisplayRepositoryImpl;
+import service.display.DisplayService;
+
+import java.util.List;
+
+public class DisplayServiceImpl implements DisplayService {
+    private DisplayRepository displayRepository = new DisplayRepositoryImpl();
```
Comment (AndrewPolishchuc) [inline src/main/java/service/display/impl/DisplayServiceImpl.java]: The same. Same in other cases.

File: src/main/java/service/operationSystem/impl/OperationSystemServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import model.operationSystem.OperationSystem;
+import repository.operationSystem.OperationSystemRepository;
+import repository.operationSystem.impl.OperationSystemRepositoryImpl;
+import service.operationSystem.OperationSystemService;
+
+import java.util.List;
+
+public class OperationSystemServiceImpl implements OperationSystemService {
+    private OperationSystemRepository operationSystemRepository = new OperationSystemRepositoryImpl();
+    @Override
```
Comment (AndrewPolishchuc) [inline src/main/java/service/operationSystem/impl/OperationSystemServiceImpl.java:19]: new line!

File: src/main/java/service/processor/impl/ProcessorServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import model.processor.Processor;
+import repository.processor.ProcessorRepository;
+import repository.processor.impl.ProcessorRepositoryImpl;
+import service.processor.ProcessorService;
+
+import java.util.List;
+
+public class ProcessorServiceImpl implements ProcessorService {
+    private ProcessorRepository processorRepository = new ProcessorRepositoryImpl();
+    @Override
```
Comment (AndrewPolishchuc) [inline src/main/java/service/processor/impl/ProcessorServiceImpl.java:18]: the same!

**Review by AndrewPolishchuc — APPROVED** [review]
Well done!

File: src/main/java/service/connection/impl/ConnectionServiceImpl.java (hunk 2 above)
Comment (typaclever) [reply src/main/java/service/connection/impl/ConnectionServiceImpl.java]: Should i create constructor like:
public ConnectionServiceImpl(ConnectionRepository repository){
        coonnectionRepository = repository;
    }
and give it ConnectionRepositoryImpl?

File: src/main/java/Exeptions/computer/ComputerDeleteException.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package Exeptions.computer;
+
+public class ComputerDeleteException extends Exception {
+    public ComputerDeleteException() {
+    }
```
Comment (AndrewPolishchuc) [inline src/main/java/Exeptions/computer/ComputerDeleteException.java]: It's too much

File: src/main/java/Exeptions/computer/ComputerUpdateException.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package Exeptions.computer;
+
+public class ComputerUpdateException extends Exception {
+
+    public ComputerUpdateException() {
+    }
```
Comment (AndrewPolishchuc) [inline src/main/java/Exeptions/computer/ComputerUpdateException.java]: The same

File: src/main/java/repository/connection/impl/ConnectionRepositoryImpl.java [hunk 8]
```diff
@@ -1,4 +1,66 @@
...
+    }
+
+    @Override
+    public Connection update(Connection connection) throws ConnectionUpdateException {
+        Session session = SessionFactoryAccess.getSessionFactory().openSession();
+        session.beginTransaction();
+        session.update(connection);
+        try {
+            session.getTransaction().commit();
+        }catch (OptimisticLockException e){
```
Comment (AndrewPolishchuc) [inline src/main/java/repository/connection/impl/ConnectionRepositoryImpl.java:31]: space

**Review by AndrewPolishchuc — APPROVED** [review]

**Pull request merged**